simple-navmesh-constraint.js

```diff
@@ -21,7 +21,28 @@ AFRAME.registerComponent('simple-navmesh-constraint', {
       default: ''
     }
   },
-  
+
+  init: function () {
+    this.updateNavmeshEntities = this.updateNavmeshEntities.bind(this);
+  },
+
+  play: function () {
+    this.el.sceneEl.addEventListener('child-attached', this.updateNavmeshEntities);
+    this.el.sceneEl.addEventListener('child-detached', this.updateNavmeshEntities);
+  },
+
+  pause: function () {
+    this.el.sceneEl.removeEventListener('child-attached', this.updateNavmeshEntities);
+    this.el.sceneEl.removeEventListener('child-detached', this.updateNavmeshEntities);
+  },
+
+  updateNavmeshEntities: function (evt) {
+    // Don't bother updating if the entity is not relevant to us
+    if (evt.detail.el.matches(`${this.data.navmesh}, ${this.data.exclude}`)) {
+      this.update();
+    }
+  },
+
```

Comment:
Not in your code, but just seeing the line 50 below, it seems `els === null` is wrong, it's not possible, it should probably be `els.length === 0` here, and in this case, setting `this.objects = els` is good and would avoid creating another array. This can be fixed in another PR.

Reply:
This was addressed in #28